test/rsa_sp800_56b_test.c

```diff
@@ -300,9 +300,17 @@ static int test_check_private_exponent(void)
           && TEST_true(BN_set_word(d, 157))
           && TEST_true(BN_set_word(n, 15*17))
           && TEST_true(RSA_set0_factors(key, p, q))
-          && TEST_true(RSA_set0_key(key, n, e, d))
-          /* fails since d >= lcm(p-1, q-1) */
-          && TEST_false(rsa_check_private_exponent(key, 8, ctx))
+          && TEST_true(RSA_set0_key(key, n, e, d));
+    if (!ret) {
+        BN_free(p);
+        BN_free(q);
```

Comment:
So if the `RSA_set0_factors` call succeeds but `RSA_set0_key` fails, doesn't this mean we're going to double free `p` and `q` because the ownership has already been transferred to `key`?

Reply:
well spotted. thanks